source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -3545,6 +3545,242 @@ export class ChoiceSetInput extends Input {
+				for (let j = 0; j <= ratingClicked; j++) {
+					labelElements[j].style.backgroundImage = "url('" + iconSelected + "')";
+				}
+				for (let j = ratingClicked + 1; j < maxValue; j++) {
+					labelElements[j].style.backgroundImage = "url('" + iconUnselected + "')";
+				}
+			};
+
+			let spacerElement = document.createElement("div");
+			spacerElement.style.width = "6px";
+
+			let compoundInput = document.createElement("div");
+			compoundInput.style.marginLeft = "0px";
+			compoundInput.style.marginRight = "0px";
+			compoundInput.style.display = "inline-block";
+			compoundInput.style.textAlign = "center";
+			compoundInput.style.flexGrow = "1";
+
+			Utils.appendChild(compoundInput, radioInput);
+			Utils.appendChild(compoundInput, spacerElement);
+			Utils.appendChild(compoundInput, labelElements[i]);
+
+			Utils.appendChild(element, compoundInput);
+
+		}
+
+		return element;
+	}
+	
+	wrap: boolean = false;
+	maxValue: number;
+	iconSelected: string;
+	iconUnselected: string;
+
+	private newMethod() {
+		return this;
+	}
+
+    getJsonTypeName(): string {
+        return "Input.Rating";
+    }
+
+    toJSON() {
+        let result = super.toJSON();
+
+		if (this.maxValue > 0) {
+            var ratings = [];
+
+			for (let i = 0; i < this.maxValue; i++) {
+				let rating: Choice;
+				rating.title = "Choice " + (i + 1);
+				rating.value = (i + 1).toString();
+				ratings.push(rating.toJSON());
+			}
+
+            Utils.setProperty(result, "choices", ratings);
+		}
+		
+        Utils.setProperty(result, "wrap", this.wrap, false);
+		
+        return result;
+    }
+
+    internalValidateProperties(context: ValidationResults) {
+		super.internalValidateProperties(context);
+		
+		const MIN_RATING_COUNT: number = 1;
+
+		if (this.maxValue < MIN_RATING_COUNT) {
+            context.addFailure(
+                this,
+                {
+					error: Enums.ValidationError.InvalidPropertyValue,
+                    message: "An Input.Rating must have at least " + MIN_RATING_COUNT + " possible rating(s)."
+                });
+        }
+    }
+
+    parse(json: any, errors?: Array<HostConfig.IValidationError>) {
+		super.parse(json, errors);
+		
+		this.maxValue = parseInt(Utils.getStringValue(json["maxValue"]), 10);
+		this.iconSelected = Utils.getStringValue(json["iconSelected"]);
+		this.iconUnselected = Utils.getStringValue(json["iconUnselected"]);
+
+		let ratings: Array<Choice> = [];
+
+		if (json["maxValue"] != undefined && json["maxValue"] > 0) {
+
+            for (let i = 0; i < json["maxValue"]; i++) {
+                let rating = new Choice();
+				rating.title = "Rating " + (i + 1);
+				rating.value = (i + 1).toString();
+
+				ratings.push(rating);
+            }
+		}
+
+        this.wrap = Utils.getBoolValue(json["wrap"], this.wrap);
+    }
+
+	get value(): string {
+		if (!this._toggleInputs || this._toggleInputs.length == 0) {
+			return null;
+		}
+
+		for (var i = 0; i < this._toggleInputs.length; i++) {
```

Comment:
You could store the current value as the radio inputs are clicked into a private member and return the private member here, avoiding that loop.